### DISCUSSION TITLE
Optional HttpUrl with alias

### DISCUSSION THREAD

**Discussion by xydian** [pydantic/pydantic#4490] [Question] [answered]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this is a bug
- [X] I am confident that the issue is with pydantic  (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

Hi :) 

I am unable to create a field with an optional `HttpUrl`. I am trying like this: 

`website: Optional[HttpUrl] = Field(..., alias='Website')`

If i run the code and do not provide a url I get the following error in postman as a response. 

```
{
    "detail": [
        {
            "loc": [
                "body",
                "Website"
            ],
            "msg": "ensure this value has at least 1 characters",
            "type": "value_error.any_str.min_length",
            "ctx": {
                "limit_value": 1
            }
        }
    ]
}
```

If I use `str` instead of `HttpUrl` everything works as expected. Is there a way to add an optional URL that is type checked? 

### Example Code

```Python
from typing import Optional

from pydantic import HttpUrl, EmailStr, BaseModel, SecretStr, Field

from fastapi import FastAPI, APIRouter


class UserCreate(BaseModel):
    email: EmailStr = Field(..., alias='Email')
    username: str = Field(..., alias='DisplayName')
    password: SecretStr = Field(..., alias='Password')
    website: Optional[HttpUrl] = Field(..., alias='Website')


router = APIRouter(prefix="/test", tags=["test"])


@router.post("/")
async def post_create_user(user: UserCreate):
    pass

app = FastAPI()


app.include_router(router)


if __name__ == '__main__':
    import uvicorn

    uvicorn.run(app, host="0.0.0.0", port=8000)
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.2
            pydantic compiled: True
                 install path: /Users/chris/Development/incenty-dirs/server_python/venv/lib/python3.8/site-packages/pydantic
               python version: 3.8.9 (default, May 17 2022, 12:55:41)  [Clang 13.1.6 (clang-1316.0.21.2.5)]
                     platform: macOS-12.4-arm64-arm-64bit
     optional deps. installed: ['dotenv', 'email-validator', 'typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [X] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [X] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.dict()` and `.json()`
- [X] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [ ] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Settings Management](https://pydantic-docs.helpmanual.io/usage/settings/)
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by samuelcolvin** [comment, marked as answer]
That's because the field is not optional - e.g. is required.

Use

```py
website: Optional[HttpUrl] = Field(None, alias='Website')
```
Notice the first argument to `Field` is `None`, not an ellipsis which means "required".

**Comment by xydian** [comment]
Thanks for the fast reply! I tried it also setting it to None but that doesn't work either. Same error message

**Reply by xydian** [reply]
@samuelcolvin can you please reopen the issue?

**Reply by samuelcolvin** [reply]
This isn't an issue it's a discussion, it is open. 

I assure, optional fields work. 

If you really think there's a bug, please provide a complete example and expected behavior.

**Reply by samuelcolvin** [reply]
And remove the fastapi code and just use pydantic to keep it simple.
